nautobot_golden_config/forms.py

```diff
@@ -598,3 +598,18 @@ class Meta:
             "change_control_url",
             "tags",
         ]
+
+
+class GenerateIntendedConfigForm(django_forms.Form):
+    """Form for generating intended configuration."""
+
+    device = forms.DynamicModelChoiceField(
+        queryset=Device.objects.all(),
+        required=True,
+        label="Device",
+    )
+    git_repository = forms.DynamicModelChoiceField(
+        queryset=GitRepository.objects.all(),
+        required=True,
```

Comment:
Just thinking out loud.. can we make this not required and if not set figure out what the default git repo is this device is?

Reply:
In the majority of cases, I don't think people will need to switch this.

Reply:
Because different devices can have different golden config settings, and different golden config settings can have different git repositories, we would have no way of showing the user in this view what repo was used to generate the template.

Either:
- They would have to back-solve which golden config setting applied and then which git repository
- We will have to enhance the API response to provide more information than it does today